lib/ansible/module_utils/crypto.py

```diff
@@ -63,7 +63,7 @@ def load_privatekey(path, passphrase=None):
         if passphrase:
             privatekey = crypto.load_privatekey(crypto.FILETYPE_PEM,
                                                 open(path, 'rb').read(),
-                                                passphrase)
+                                                passphrase.encode('UTF-8'))
```

Comment:
With Python 2, it fails when a non-ASCII passphrase is used:

```
TASK [openssl_privatekey : Generate privatekey6 - standard - with non ASCII passphrase] ***
[...]
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_93C9Ge/ansible_module_openssl_privatekey.py", line 301, in <module>
    main()
  File "/tmp/ansible_93C9Ge/ansible_module_openssl_privatekey.py", line 280, in main
    private_key.generate(module)
  File "/tmp/ansible_93C9Ge/ansible_module_openssl_privatekey.py", line 186, in generate
    self.cipher, self.passphrase.encode('UTF-8')))
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 0: ordinal not in range(128)

fatal: [testhost]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_93C9Ge/ansible_module_openssl_privatekey.py\", line 301, in <module>\n    main()\n  File \"/tmp/ansible_93C9Ge/ansible_module_openssl_privatekey.py\", line 280, in main\n    private_key.generate(module)\n  File \"/tmp/ansible_93C9Ge/ansible_module_openssl_privatekey.py\", line 186, in generate\n    self.cipher, self.passphrase.encode('UTF-8')))\nUnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 0: ordinal not in range(128)\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE", 
    "rc": 0
}
```

[Test case](https://github.com/pilou-/ansible/commit/470371db5e462b53a9f455e86e5fd74068fb7d64) added on this [branch](https://github.com/pilou-/ansible/commits/crypto_more_tests_2), another commit fix 2 typos in tests.